mlrun/runtimes/base.py

```diff
@@ -459,6 +459,8 @@ def run(
             results = runner(task_generator, execution, run)
             results_to_iter(results, run, execution)
             result = execution.to_dict()
+            if execution.state == RunStates.completed:
+                result = self._update_run_state(result, task=run)
```

Comment:
what happens when not completed?

Reply:
well, the state would be changed somewhere else.. but you are right this is confusing. removed the if and always call the method `_update_run_state `